src/lib/y2storage/boot_requirements_strategies/prep.rb

```diff
@@ -55,6 +64,34 @@ def warnings
 
     protected
 
+      # PReP partition is needed, so return any warning related to it.
+      def prep_warnings
+        res = []
+        big_preps = too_big_preps
+
+        if missing_partition_for?(prep_volume)
+          res << SetupError.new(missing_volume: prep_volume)
+        elsif !big_preps.empty?
+          res << SetupError.new(message: big_prep_warning(big_preps))
+        end
+
+        res
+      end
+
+      def big_prep_warning(big_partitions)
+        # TRANSLATORS: %s is single or list of partitions that are too big.
+        msg = format(
+          n_("Following PReP partition is too big: %s. ",
+            "Following PReP partitions are too big: %s.",
+            big_partitions.size
+          ),
+          big_partitions.map(&:name).join(", ")
+        )
+        # TRANSLATORS: %s is human readable partition size like 8 MiB.
+        msg += format(_("Some firmwares can refuse to load PReP partition " \
+          "that is bigger %s and thus prevent booting."), MAX_PREP_SIZE)
```

Comment:
"refuse to load PReP partition that is bigger 8 MiB" sounds strange. I would go for "refuse to load PReP partitions bigger **than** 8 MiB". Or being closer to your idea "refuse to load **any** PReP Partition that is bigger **than** 8 MiB"